ovs-dpdk.md

```diff
@@ -0,0 +1,180 @@
+# oVirt now supports OVS-DPDK
+
+
+[DPDK](http://dpdk.org/) (Data Plane Development Kit) provides high-performance packet processing libraries and user space drivers.
+[Open vSwitch](http://openvswitch.org/)  supports DPDK since v2.2 and starting with version 2.4, a DPDK-optimized vHost path is enabled in OVS.
+ 
+According to [Intel ONP performance tests](https://download.01.org/packet-processing/ONPS2.1/Intel_ONP_Release_2.1_Performance_Test_Report_Rev1.0.pdf), using OVS with DPDK can provide a huge performance enhancement, increasing network packets throughput and reduced latency.
+ 
+OVS-DPDK has been added to oVirt as an experimental feature.
+Please note that OVS-DPDK cannot access any features located within the Linux kernel, for example: Linux bridge,  tun/tap devices, iptables etc.
+
+## Requirements:
+
+Please follow the instructions in <http://dpdk.org/doc/guides-16.04/linux_gsg/nic_perf_intel_platform.html> in order to acheive best performance.
+ 
+The network card must be on the supported-vendor matrix, located here: <http://dpdk.org/doc/nics>
+
+Please review that your system supports 1GB hugepages:
+```if grep pdpe1gb /proc/cpuinfo >/dev/null 2>&1; then echo "1GB supported."; fi```
+
+* Hardware support: Make sure VT-d / AMD-Vi is enabled in BIOS
+* Kerenel support: When adding a new host via oVirt GUI, go to Hosts -> Edit > Kernel and add the following parameters to Kernel Command line:  ```iommu=pt intel_iommu=on default_hugepagesz=1GB hugepagesz=1G hugepages=N ```
+In case that 1GB hugepages is not supported, please drop ```default_hugepagesz=1GB hugepagesz=1G``` since default hugepages size will be used.
+
+It is best to configure the host before installing it on oVirt-engine, but if changes are made after vdsm is installed, move the host to maintenance mode, reboot the host and reinstall it, for the changes to take effect.
+
+
+## Configure OVS and DPDK on the Host
+
+### Install dpdk packages:
+
+```bash
+# yum install dpdk dpdk-tools driverctl
+
+```
+ 
+### Bind NICs to DPDK:
+
+In order to run a DPDK application, a suitable uio module should be loaded into kernel memory.
+Available modules are: uio_pci_generic, igb_uio and vfio-pci.
+In many cases, the standard uio_pci_generic module included in the Linux kernel can provide the uio capability.
+For some devices which lack support for legacy interrupts, igb_uio should be used, but it’s an out-of-tree kernel module.
```

Comment:
Suggest:
"For devices that lack support for legacy interrupts, igb_uio should be used. Note that it is an out-of-tree kernel module."